### ISSUE TITLE
multicast_dns "SocketException: Failed to create datagram socket" on windows

### ISSUE THREAD

**Issue by casvanluijtelaar** [flutter/flutter#90051]
## Steps to Reproduce

1. Run `flutter create bug`.
2. add `multicast_dns: ^0.3.1` to `pubspec.yaml`
3. create a new file: 
``` dart
import 'dart:io';

import 'package:multicast_dns/multicast_dns.dart';

/// class handling listening for multi_dns devices
class MdnsDeviceDiscovery {
  /// mDNS default address
  static const _name = '_http._tcp';

  /// creates the MDnsClient, we have to overwrite the reusePort because it's
  /// currently not supported on windows
  /// https://github.com/flutter/flutter/issues/55173
  final _client = MDnsClient(
      rawDatagramSocketFactory: (dynamic host, int port,
              {bool? reuseAddress, bool? reusePort, int? ttl}) =>
          RawDatagramSocket.bind(host, port, ttl: ttl!));

  /// start the MdnsDeviceDiscovery client
  Future<void> start() => _client.start();

  /// stop the MdnsDeviceDiscovery client
  void stop() => _client.stop();

  /// receive an instance of the stream containing the service records for
  /// listening to.
  Stream<SrvResourceRecord> get stream async* {
    final ptrQuery = ResourceRecordQuery.serverPointer(_name);
    final ptrRecord = _client.lookup<PtrResourceRecord>(ptrQuery);

    await for (final ptr in ptrRecord) {
      final srvQuery = ResourceRecordQuery.service(ptr.domainName);
      final srvRecord = _client.lookup<SrvResourceRecord>(srvQuery);

      yield await srvRecord.first; // only use the first service record
    }
  }
}
```
4. test it:
``` dart
void test() async {
    final _discovery = MdnsDeviceDiscovery();
    await _discovery.start();

   ....
}
``` 

**Expected results:**
I'm expecting this future to complete correctly and using the rest of the class to correctly find the devices I'm looking for. This works perfectly fine on Android.

**Actual results:** 
throws an error on  `await _discovery.start();`

``` powershell
SocketException (SocketException: Failed to create datagram socket (OS Error: The requested address is not valid in its context.
, errno = 10049), address = , port = 5353)
```

<details>
  <summary>Logs</summary>

``` powershell
flutter run --verbose
[ +115 ms] executing: [D:\Android\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[+1191 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] 4cc385b4b84ac2f816d939a49ea1f328c4e0b48e
[   +1 ms] executing: [D:\Android\flutter/] git tag --points-at 4cc385b4b84ac2f816d939a49ea1f328c4e0b48e
[ +144 ms] Exit code 0 from: git tag --points-at 4cc385b4b84ac2f816d939a49ea1f328c4e0b48e
[   +1 ms] 2.5.0
[  +15 ms] executing: [D:\Android\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +86 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [D:\Android\flutter/] git ls-remote --get-url origin
[  +67 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +184 ms] executing: [D:\Android\flutter/] git rev-parse --abbrev-ref HEAD
[  +77 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[ +120 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +8 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +108 ms] executing: D:\Android\platform-tools\adb.exe devices -l
[  +93 ms] List of devices attached
[  +10 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[  +62 ms] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +117 ms] Multiple devices found:
[  +51 ms] Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1165]
[   +4 ms] Chrome (web)      • chrome  • web-javascript • Google Chrome 93.0.4577.63
[   +1 ms] Edge (web)        • edge    • web-javascript • Microsoft Edge 93.0.961.47
[   +4 ms] [1]: Windows (windows)
[   +1 ms] [2]: Chrome (chrome)
[   +1 ms] [3]: Edge (edge)
[   +3 ms] Please choose one (To quit, press "q/Q")
[        ] : 
[+3883 ms] 1
[  +20 ms] Skipping pub get: version match.
[ +100 ms] Found plugin path_provider_linux at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\path_provider_linux-2.0.2\
[   +9 ms] Found plugin path_provider_windows at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\path_provider_windows-2.0.3\
[  +16 ms] Found plugin shared_preferences at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences-2.0.7\
[   +3 ms] Found plugin shared_preferences_linux at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_linux-2.0.2\
[   +4 ms] Found plugin shared_preferences_macos at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_macos-2.0.2\
[   +4 ms] Found plugin shared_preferences_web at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_web-2.0.2\
[   +2 ms] Found plugin shared_preferences_windows at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_windows-2.0.2\
[ +326 ms] Found plugin path_provider_linux at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\path_provider_linux-2.0.2\
[   +3 ms] Found plugin path_provider_windows at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\path_provider_windows-2.0.3\
[  +11 ms] Found plugin shared_preferences at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences-2.0.7\
[   +3 ms] Found plugin shared_preferences_linux at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_linux-2.0.2\
[   +2 ms] Found plugin shared_preferences_macos at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_macos-2.0.2\
[   +3 ms] Found plugin shared_preferences_web at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_web-2.0.2\
[   +2 ms] Found plugin shared_preferences_windows at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_windows-2.0.2\
[  +31 ms] Generating D:\werk\wled_app\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[ +139 ms] Initializing file store
[  +18 ms] Skipping target: gen_localizations
[   +7 ms] Skipping target: gen_dart_plugin_registrant
[   +2 ms] Skipping target: _composite
[   +6 ms] complete
[   +7 ms] Launching lib\main.dart on Windows in debug mode...
[   +9 ms] D:\Android\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev D:\Android\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root
D:\Android\flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dillC:\Users\casva\AppData\Local\Temp\flutter_tools.ac0d3b8e\flutter_tool.8fb08de0\app.dill --packages D:\werk\wled_app\.dart_tool\package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false
--enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill build\c075001b96339384a97db4862b8ab8db.cache.dill.track.dill
--enable-experiment=alternative-invalidation-strategy
[  +42 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop
Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[  +45 ms] Exit code 0 from: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop
Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[   +1 ms] []
[        ] executing: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.VCTools
Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[  +38 ms] Exit code 0 from: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.VCTools
Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[   +1 ms] [
             {
               "instanceId": "b6730f3e",
               "installDate": "2021-09-02T13:19:33Z",
               "installationName": "VisualStudio/16.11.2+31624.102",
               "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools",
               "installationVersion": "16.11.31624.102",
               "productId": "Microsoft.VisualStudio.Product.BuildTools",
               "productPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\Common7\\Tools\\LaunchDevCmd.bat",
               "state": 4294967295,
               "isComplete": true,
               "isLaunchable": true,
               "isPrerelease": false,
               "isRebootRequired": false,
               "displayName": "Visual Studio Build Tools 2019",
               "description": "The Visual Studio Build Tools allows you to build native and managed MSBuild-based applications without requiring the Visual Studio IDE. There are options to install the Visual 
               C++ compilers and libraries, MFC, ATL, and C++/CLI support.",
               "channelId": "VisualStudio.16.Release",
               "channelUri": "https://aka.ms/vs/16/release/channel",
               "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
               "releaseNotes": "https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-v16.11#16.11.2",
               "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660909",
               "updateDate": "2021-09-02T13:19:33.0945434Z",
               "catalog": {
                 "buildBranch": "d16.11",
                 "buildVersion": "16.11.31624.102",
                 "id": "VisualStudio/16.11.2+31624.102",
                 "localBuild": "build-lab",
                 "manifestName": "VisualStudio",
                 "manifestType": "installer",
                 "productDisplayVersion": "16.11.2",
                 "productLine": "Dev16",
                 "productLineVersion": "2019",
                 "productMilestone": "RTW",
                 "productMilestoneIsPreRelease": "False",
                 "productName": "Visual Studio",
                 "productPatchVersion": "2",
                 "productPreReleaseMilestoneSuffix": "1.0",
                 "productSemanticVersion": "16.11.2+31624.102",
                 "requiredEngineVersion": "2.11.35.61819"
               },
               "properties": {
                 "campaignId": "",
                 "channelManifestId": "VisualStudio.16.Release/16.11.2+31624.102",
                 "nickname": "",
                 "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\setup.exe"
               }
             }
           ]
[  +25 ms] Building Windows application...
[  +26 ms] <- compile file:///D:/werk/wled_app/.dart_tool/flutter_build/generated_main.dart
[   +2 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe -S D:\werk\wled_app\windows -B build\windows -G Visual    
Studio 16 2019
[ +134 ms] -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19042.
[  +37 ms] -- Configuring done
[ +259 ms] -- Generating done
[  +29 ms] -- Build files have been written to: D:/werk/wled_app/build/windows
[   +9 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe --build build\windows --config Debug --target INSTALL
--verbose
[ +312 ms] Microsoft (R) Build Engine version 16.11.0+0538acc04 for .NET Framework
[        ] Copyright (C) Microsoft Corporation. All rights reserved.
[  +95 ms] Build started 14/09/2021 15:28:15.
[ +352 ms] Project "D:\werk\wled_app\build\windows\INSTALL.vcxproj" on node 1 (default targets).
[   +1 ms] Project "D:\werk\wled_app\build\windows\INSTALL.vcxproj" (1) is building "D:\werk\wled_app\build\windows\ZERO_CHECK.vcxproj" (2) on node 1 (default targets).
[  +14 ms] InitializeBuildStatus:
[   +4 ms]   Creating "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +34 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +51 ms] FinalizeBuildStatus:
[   +1 ms]   Deleting file "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
[   +6 ms]   Touching "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate".
[   +1 ms] Done Building Project "D:\werk\wled_app\build\windows\ZERO_CHECK.vcxproj" (default targets).
[ +178 ms] Project "D:\werk\wled_app\build\windows\INSTALL.vcxproj" (1) is building "D:\werk\wled_app\build\windows\ALL_BUILD.vcxproj" (3) on node 1 (default targets).
[   +9 ms] Project "D:\werk\wled_app\build\windows\ALL_BUILD.vcxproj" (3) is building "D:\werk\wled_app\build\windows\flutter\flutter_wrapper_app.vcxproj" (4) on node 1 (default targets).
[   +1 ms] Project "D:\werk\wled_app\build\windows\flutter\flutter_wrapper_app.vcxproj" (4) is building "D:\werk\wled_app\build\windows\flutter\flutter_assemble.vcxproj" (5) on node 1 (default targets).      
[  +10 ms] InitializeBuildStatus:
[   +9 ms]   Creating "x64\Debug\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +12 ms] CustomBuild:
[   +2 ms]   "The build of 'D:\werk\wled_app\build\windows\CMakeFiles\e52027d35f6b1d74163fd0c1a1761e00\flutter_assemble.rule' depends on 'D:\WERK\WLED_APP\WINDOWS\FLUTTER\EPHEMERAL\FLUTTER_WINDOWS.DLL' which 
is produced by the build of 'D:\werk\wled_app\build\windows\CMakeFiles\03d3434ca014292dc55f5ad6776dbf34\flutter_windows.dll.rule'. The items cannot be built in parallel."
[ +112 ms]   Generating D:/werk/wled_app/windows/flutter/ephemeral/flutter_windows.dll, D:/werk/wled_app/windows/flutter/ephemeral/flutter_export.h,
D:/werk/wled_app/windows/flutter/ephemeral/flutter_windows.h, D:/werk/wled_app/windows/flutter/ephemeral/flutter_messenger.h, D:/werk/wled_app/windows/flutter/ephemeral/flutter_plugin_registrar.h,
D:/werk/wled_app/windows/flutter/ephemeral/flutter_texture_registrar.h, D:/werk/wled_app/windows/flutter/ephemeral/cpp_client_wrapper/core_implementations.cc,
D:/werk/wled_app/windows/flutter/ephemeral/cpp_client_wrapper/standard_codec.cc, D:/werk/wled_app/windows/flutter/ephemeral/cpp_client_wrapper/plugin_registrar.cc,
D:/werk/wled_app/windows/flutter/ephemeral/cpp_client_wrapper/flutter_engine.cc, D:/werk/wled_app/windows/flutter/ephemeral/cpp_client_wrapper/flutter_view_controller.cc, _phony_
[+2482 ms]   [  +96 ms] executing: [D:\Android\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +650 ms]   [ +677 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +2 ms]   [   +1 ms] 4cc385b4b84ac2f816d939a49ea1f328c4e0b48e
[  +16 ms]   [        ] executing: [D:\Android\flutter/] git tag --points-at 4cc385b4b84ac2f816d939a49ea1f328c4e0b48e
[ +126 ms]   [ +144 ms] Exit code 0 from: git tag --points-at 4cc385b4b84ac2f816d939a49ea1f328c4e0b48e
[   +1 ms]   [        ] 2.5.0
[  +14 ms]   [  +10 ms] executing: [D:\Android\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +78 ms]   [  +82 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ]   [        ] origin/stable
[   +7 ms]   [        ] executing: [D:\Android\flutter/] git ls-remote --get-url origin
[  +71 ms]   [  +78 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms]   [        ] https://github.com/flutter/flutter.git
[  +78 ms]   [  +80 ms] executing: [D:\Android\flutter/] git rev-parse --abbrev-ref HEAD
[  +94 ms]   [  +94 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms]   [        ] stable
[  +84 ms]   [  +85 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +2 ms]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +4 ms]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[  +13 ms]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms]   [   +8 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +5 ms]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +2 ms]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +132 ms]   [ +160 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[   +3 ms]   [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[  +12 ms]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[   +1 ms]   [   +5 ms] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +2 ms]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +3 ms]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[   +2 ms]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[  +17 ms]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[   +1 ms]   [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +40 ms]   [  +88 ms] Initializing file store
[  +36 ms]   [  +36 ms] Skipping target: gen_localizations
[  +34 ms]   [  +34 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents: D:\werk\wled_app\.dart_tool\package_config_subset}
[   +2 ms]   [   +2 ms] unpack_windows: Starting due to {}
[  +15 ms]   [  +15 ms] Found plugin shared_preferences at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences-2.0.7\
[   +7 ms]   [   +2 ms] Found plugin shared_preferences_linux at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_linux-2.0.2\
[   +8 ms]   [   +3 ms] Found plugin shared_preferences_macos at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_macos-2.0.2\
[  +13 ms]   [   +3 ms] Found plugin shared_preferences_web at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_web-2.0.2\
[   +5 ms]   [   +2 ms] Found plugin shared_preferences_windows at C:\Users\casva\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\shared_preferences_windows-2.0.2\
[  +43 ms]   [  +64 ms] gen_dart_plugin_registrant: Complete
[  +18 ms]   [   +5 ms] kernel_snapshot: Starting due to {}
[   +3 ms]   [  +15 ms] D:\Android\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev D:\Android\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root
D:\Android\flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --target=flutter --no-print-incremental-dependencies -DFLUTTER_WEB_AUTO_DETECT=true -Ddart.vm.profile=false -Ddart.vm.product=false                                                                                                                                                                       k-root
--enable-asserts --track-widget-creation --packages D:\werk\wled_app\.dart_tool\package_config.json --output-dill D:\werk\wled_app\.dart_tool\flutter_build\5469780880d59144619fcbb3ba842drt.vm.product=false   1d\app.dill --depfile                                                                                                                                                                     1d\app.dill --depfile 
D:\werk\wled_app\.dart_tool\flutter_build\5469780880d59144619fcbb3ba842d1d\kernel_snapshot.d D:\werk\wled_app\.dart_tool\flutter_build\generated_main.dart
[+16006 ms]   [+16007 ms] kernel_snapshot: Complete
[+1057 ms]   [+1059 ms] debug_bundle_windows_assets: Starting due to {}
[ +225 ms]   [ +226 ms] debug_bundle_windows_assets: Complete
[ +822 ms]   [ +822 ms] Persisting file store
[  +17 ms]   [  +17 ms] Done persisting file store
[  +14 ms]   [  +14 ms] build succeeded.
[  +17 ms]   [  +17 ms] "flutter assemble" took 23,439ms.
[  +79 ms]   [  +80 ms] ensureAnalyticsSent: 74ms
[   +1 ms]   [   +1 ms] Running shutdown hooks
[   +1 ms]   [        ] Shutdown hooks complete
[        ]   [        ] exiting with code 0
[  +96 ms] FinalizeBuildStatus:
[   +1 ms]   Deleting file "x64\Debug\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\flutter_assemble\flutter_assemble.tlog\flutter_assemble.lastbuildstate".
[        ] Done Building Project "D:\werk\wled_app\build\windows\flutter\flutter_assemble.vcxproj" (default targets).
[   +7 ms] InitializeBuildStatus:
[   +1 ms]   Creating "flutter_wrapper_app.dir\Debug\flutter_.CA75BB77.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +1 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +70 ms] ClCompile:
[        ]   All outputs are up-to-date.
[  +30 ms] Lib:
[        ]   All outputs are up-to-date.
[   +1 ms]   flutter_wrapper_app.vcxproj -> D:\werk\wled_app\build\windows\flutter\Debug\flutter_wrapper_app.lib
[  +13 ms] FinalizeBuildStatus:
[        ]   Deleting file "flutter_wrapper_app.dir\Debug\flutter_.CA75BB77.tlog\unsuccessfulbuild".
[        ]   Touching "flutter_wrapper_app.dir\Debug\flutter_.CA75BB77.tlog\flutter_wrapper_app.lastbuildstate".
[        ] Done Building Project "D:\werk\wled_app\build\windows\flutter\flutter_wrapper_app.vcxproj" (default targets).
[  +45 ms] Project "D:\werk\wled_app\build\windows\ALL_BUILD.vcxproj" (3) is building "D:\werk\wled_app\build\windows\flutter\flutter_wrapper_plugin.vcxproj" (6) on node 1 (default
targets).
[   +1 ms] InitializeBuildStatus:
[        ]   Creating "flutter_wrapper_plugin.dir\Debug\flutter_.9F156331.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +1 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +12 ms] ClCompile:
[        ]   All outputs are up-to-date.
[   +5 ms] Lib:
[        ]   All outputs are up-to-date.
[        ]   flutter_wrapper_plugin.vcxproj -> D:\werk\wled_app\build\windows\flutter\Debug\flutter_wrapper_plugin.lib
[   +9 ms] FinalizeBuildStatus:
[   +1 ms]   Deleting file "flutter_wrapper_plugin.dir\Debug\flutter_.9F156331.tlog\unsuccessfulbuild".
[        ]   Touching "flutter_wrapper_plugin.dir\Debug\flutter_.9F156331.tlog\flutter_wrapper_plugin.lastbuildstate".
[        ] Done Building Project "D:\werk\wled_app\build\windows\flutter\flutter_wrapper_plugin.vcxproj" (default targets).
[  +49 ms] Project "D:\werk\wled_app\build\windows\ALL_BUILD.vcxproj" (3) is building "D:\werk\wled_app\build\windows\runner\wled_app.vcxproj" (7) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "wled_app.dir\Debug\wled_app.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +2 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +50 ms] ClCompile:
[   +1 ms]   C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\CL.exe /c /ID:\werk\wled_app\windows
/ID:\werk\wled_app\windows\flutter\ephemeral /ID:\werk\wled_app\windows\flutter\ephemeral\cpp_client_wrapper\include /Zi /nologo /W4 /WX /diagnostics:column /Od /Ob0 /D WIN32 /D _WINDOWS/D _HAS_EXCEPTIONS=0 /D _DEBUG /D NOMINMAX /D UNICODE /D _UNICODE /D "CMAKE_INTDIR=\"Debug\"" /D _UNICODE /D UNICODE /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope       
/Zc:inline /GR /std:c++17 /Fo"wled_app.dir\Debug\\" /Fd"wled_app.dir\Debug\vc142.pdb" /external:W4 /Gd /TP /wd4100 /errorReport:queue
D:\werk\wled_app\windows\flutter\generated_plugin_registrant.cc D:\werk\wled_app\windows\runner\flutter_window.cpp
[ +150 ms]   generated_plugin_registrant.cc
[ +606 ms]   flutter_window.cpp
[+1152 ms]   Generating Code...
[ +205 ms] ResourceCompile:
[   +1 ms]   All outputs are up-to-date.
[ +186 ms] Link:
[   +2 ms]   C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\link.exe /ERRORREPORT:QUEUE
/OUT:"D:\werk\wled_app\build\windows\runner\Debug\wled_app.exe" /INCREMENTAL /ILK:"wled_app.dir\Debug\wled_app.ilk" /NOLOGO ..\flutter\Debug\flutter_wrapper_app.lib
D:\werk\wled_app\windows\flutter\ephemeral\flutter_windows.dll.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib   
/MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /manifestinput:D:\werk\wled_app\windows\runner\runner.exe.manifest /DEBUG
/PDB:"D:/werk/wled_app/build/windows/runner/Debug/wled_app.pdb" /SUBSYSTEM:WINDOWS /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"D:/werk/wled_app/build/windows/runner/Debug/wled_app.lib"     
/MACHINE:X64  /machine:x64 wled_app.dir\Debug\Runner.res
[   +2 ms]   wled_app.dir\Debug\flutter_window.obj
[        ]   wled_app.dir\Debug\main.obj
[   +1 ms]   wled_app.dir\Debug\utils.obj
[        ]   wled_app.dir\Debug\win32_window.obj
[        ]   wled_app.dir\Debug\generated_plugin_registrant.obj
[ +606 ms]   wled_app.vcxproj -> D:\werk\wled_app\build\windows\runner\Debug\wled_app.exe
[   +9 ms] FinalizeBuildStatus:
[   +2 ms]   Deleting file "wled_app.dir\Debug\wled_app.tlog\unsuccessfulbuild".
[   +2 ms]   Touching "wled_app.dir\Debug\wled_app.tlog\wled_app.lastbuildstate".
[   +2 ms] Done Building Project "D:\werk\wled_app\build\windows\runner\wled_app.vcxproj" (default targets).
[   +6 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +22 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +10 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild".
[   +1 ms]   Touching "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\ALL_BUILD.lastbuildstate".
[        ] Done Building Project "D:\werk\wled_app\build\windows\ALL_BUILD.vcxproj" (default targets).
[   +6 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +80 ms] PostBuildEvent:
[   +1 ms]   setlocal
[        ]   "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DBUILD_TYPE=Debug -P cmake_install.cmake  
[        ]   if %errorlevel% neq 0 goto :cmEnd
[   +2 ms]   :cmEnd
[   +1 ms]   endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
[        ]   :cmErrorLevel
[   +1 ms]   exit /b %1
[   +1 ms]   :cmDone
[   +1 ms]   if %errorlevel% neq 0 goto :VCEnd
[   +1 ms]   :VCEnd
[  +84 ms]   -- Install configuration: "Debug"
[  +24 ms]   -- Up-to-date: D:/werk/wled_app/build/windows/runner/Debug/data/icudtl.dat
[        ]   -- Up-to-date: D:/werk/wled_app/build/windows/runner/Debug/flutter_windows.dll
[  +11 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets
[   +3 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/AssetManifest.json
[   +3 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/FontManifest.json
[   +2 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/fonts
[        ]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/fonts/MaterialIcons-Regular.otf
[   +6 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/kernel_blob.bin
[ +181 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/NOTICES.Z
[   +2 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/packages
[   +5 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/packages/flutter_feather_icons
[   +1 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/packages/flutter_feather_icons/fonts
[   +3 ms]   -- Installing: D:/werk/wled_app/build/windows/runner/Debug/data/flutter_assets/packages/flutter_feather_icons/fonts/feather.ttf
[  +18 ms] FinalizeBuildStatus:
[   +3 ms]   Deleting file "x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild".
[   +1 ms]   Touching "x64\Debug\INSTALL\INSTALL.tlog\INSTALL.lastbuildstate".
[   +1 ms] Done Building Project "D:\werk\wled_app\build\windows\INSTALL.vcxproj" (default targets).
[   +2 ms] Build succeeded.
[        ]     0 Warning(s)
[   +1 ms]     0 Error(s)
[        ] Time Elapsed 00:00:31.87
[  +21 ms] Building Windows application... (completed in 32.8s)
[ +911 ms] Observatory URL on device: http://127.0.0.1:50604/haqLPlFkLkM=/
[   +7 ms] Caching compiled dill
[ +150 ms] Connecting to service protocol: http://127.0.0.1:50604/haqLPlFkLkM=/
[ +190 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:50604/haqLPlFkLkM=/.
[ +117 ms] DDS is listening at http://127.0.0.1:50609/5gffF00vNe0=/.
[  +82 ms] Successfully connected to service protocol: http://127.0.0.1:50604/haqLPlFkLkM=/
[  +20 ms] DevFS: Creating new filesystem on the device (null)
[  +19 ms] DevFS: Created new filesystem on the device (file:///C:/Users/casva/AppData/Local/Temp/wled_appe0ffba4/wled_app/)
[   +2 ms] Updating assets
[ +244 ms] Syncing files to device Windows...
[   +8 ms] <- reset
[   +1 ms] Compiling dart to kernel with 0 updated files
[  +11 ms] <- recompile file:///D:/werk/wled_app/.dart_tool/flutter_build/generated_main.dart 5eb24aa8-b739-45cd-b426-7f480ce1d552
[   +1 ms] <- 5eb24aa8-b739-45cd-b426-7f480ce1d552
[ +225 ms] Updating files.
[   +3 ms] DevFS: Sync finished
[  +52 ms] Syncing files to device Windows... (completed in 287ms)
[  +28 ms] Synced 0.0MB.
[   +4 ms] <- accept
[  +33 ms] Connected to _flutterView/0x221127b9950.
[  +10 ms] Flutter run key commands.
[   +2 ms] r Hot reload. 
[   +1 ms] R Hot restart.
[   +1 ms] h List all available interactive commands.
[   +2 ms] d Detach (terminate "flutter run" but leave application running).
[   +1 ms] c Clear the screen
[   +1 ms] q Quit (terminate the application on the device).
[   +3 ms]  Running with sound null safety 
[   +2 ms] An Observatory debugger and profiler on Windows is available at: http://127.0.0.1:50609/5gffF00vNe0=/
[ +882 ms] Activating Dart DevTools...
[+8187 ms] Activating Dart DevTools... (completed in 8.2s)
[ +711 ms] The Flutter DevTools debugger and profiler on Windows is available at: http://127.0.0.1:9102?uri=http://127.0.0.1:50609/5gffF00vNe0=/
```

``` powershell
flutter analyze
Analyzing wled_app...                                                   
No issues found! (ran in 5.9s)
```

``` powershell
flutter doctor -v
[√] Flutter (Channel stable, 2.5.0, on Microsoft Windows [Version 10.0.19042.1165], locale en-GB)
    • Flutter version 2.5.0 at D:\Android\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (6 days ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at D:\Android\
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = D:\Android
    • Java binary at: D:\Android\openjdk\bin\java
    • Java version OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_202-b08)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
    • Visual Studio Build Tools 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1165]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 93.0.4577.63
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 93.0.961.47
```

</details>

**Comment by darshankawar** [comment]
@casvanluijtelaar 
I believe this is a duplicate of https://github.com/flutter/flutter/issues/53477 and you seem to have commented on it with your findings.
Are you fine to track this in original issue ?

**Comment by casvanluijtelaar** [comment]
I already forgot about that one, yep!
